lib/mixpanel/uri.rb

```diff
@@ -15,7 +18,13 @@ def self.mixpanel(resource, params)
     end
 
     def self.encode(params)
-      params.map { |key, val| "#{key}=#{CGI.escape(val.to_s)}" }.sort.join('&')
+      params.map do |key, val|
+        if val.is_a?(Hash) || val.is_a?(Array)
```

Comment:
The only thing that can be DRY'd up is the if statement and I don't think it's worth the trouble. You can't escape the values of the signature.

Would you like me to make a pull-request to your repository so it can support formulas?